idaes/core/surrogate/pysmo/tests/test_sampling.py

```diff
@@ -2240,5 +2241,631 @@ def test_sample_points_02(self, array_type):
+            CSClass = CustomSampling(
+                input_array,
+                number_of_samples=num_samples,
+                sampling_type="selection",
+                list_of_distributions=["uniform", "normal"],
+            )
+            dist_type = "uniform"
+            dist_res, scaled_samples = CSClass.generate_from_dist(dist_type)
+            assert type(scaled_samples) == np.ndarray
+            assert scaled_samples.shape == (CSClass.number_of_samples,)
+            assert dist_res.__name__ == dist_type
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("array_type", [np.array])
+    def test_generate_from_dist_02(self, array_type):
+        for num_samples in [None, 10, 1]:
+            input_array = array_type(self.input_array)
+            CSClass = CustomSampling(
+                input_array,
+                number_of_samples=num_samples,
+                sampling_type="selection",
+                list_of_distributions=["uniform", "normal"],
+            )
+            dist_type = "normal"
+            dist_res, scaled_samples = CSClass.generate_from_dist(dist_type)
+            assert type(scaled_samples) == np.ndarray
+            assert scaled_samples.shape == (CSClass.number_of_samples,)
+            assert dist_res.__name__ == dist_type
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("array_type", [np.array])
+    def test_generate_from_dist_03(self, array_type):
+        for num_samples in [None, 10, 1]:
+            input_array = array_type(self.input_array)
+            CSClass = CustomSampling(
+                input_array,
+                number_of_samples=num_samples,
+                sampling_type="selection",
+                list_of_distributions=["uniform", "normal"],
+            )
+            dist_type = "random"
+            dist_res, scaled_samples = CSClass.generate_from_dist(dist_type)
+            assert type(scaled_samples) == np.ndarray
+            assert scaled_samples.shape == (CSClass.number_of_samples,)
+            assert dist_res.__name__ == dist_type
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("array_type", [np.array])
+    def test_generate_from_dist_04(self, array_type):
+        for num_samples in [None, 10, 1]:
+            input_array = array_type(self.input_array)
+            CSClass = CustomSampling(
+                input_array,
+                number_of_samples=num_samples,
+                sampling_type="selection",
+                list_of_distributions=["uniform", "normal"],
+            )
+            for dist_type in ["uniform", "random", "uniform"]:
+                dist_res, scaled_samples = CSClass.generate_from_dist(dist_type)
+                assert type(scaled_samples) == np.ndarray
+                assert scaled_samples.shape == (CSClass.number_of_samples,)
+                assert dist_res.__name__ == dist_type
+
+    @pytest.mark.unit
+    @pytest.mark.parametrize("array_type", [np.array])
+    def test_sample_points_01(self, array_type):
```

Comment:
These look like tests that could (should) be done using fixed inputs. I.e. do not use the sample generation and mock up a set of known distributions that we can then use to ensure we get the right answer out.

As far as I can tell, `sample_points` is a case where we can test the method in isolation used manufactured data - i.e. provide  the input specification, a mock up of a random dataset and a concrete answer we expect to get back out.